src/main.rs

```diff
@@ -135,14 +135,14 @@ enum Format {
 }
 
 #[derive(Clone, Debug)]
-struct Switch {
+struct MatchTree {
```

Comment:
Maybe some other options could be `LookaheadTree` or `ByteTree`?

Reply:
Yes, I think we're going to need other lookahead mechanisms sooner or later and might adjust names further then.